File: mobile/test/quicktag_source_names_test.dart
import 'dart:convert';
import 'dart:io';
import 'package:flutter_test/flutter_test.dart';
import 'package:novelai_mobile/i18n/app_locales.dart';
import 'package:novelai_mobile/services/online_gallery_service.dart';
import 'package:novelai_mobile/services/quicktag_ui.dart';

void main() {
  test('QuickTagCloud keeps its gallery adapter and separate source identity',
      () {
    expect(OnlineGallerySource.quicktag.label, 'QuickTagCloud');
    expect(OnlineGallerySource.quicktag.siteUrl,
        'https://novelai.quicktagcloud.com');
    final implementation =
        File('lib/services/prompt_codex_service.dart').readAsStringSync();
    expect(implementation, contains('https://nai4.top'));
    expect(implementation, isNot(contains('quicktagcloud.com')));
    final gallery =
        File('lib/screens/online_gallery_screen.dart').readAsStringSync();
    expect(gallery, isNot(contains('QuickTagCloud ·')));
  });
  test('five-language labels match desktop and do not add a main destination',
      () {
    final shared =
        jsonDecode(File('../shared/quicktag-ui.json').readAsStringSync())
            as Map;
    for (final locale in supportedAppLocales) {
      expect(quickTagUi(locale.code), shared[locale.code]);
      expect(mobileUiTextFor(locale.code, 'promptCodex.enabled'),
          contains('nai4.top'));
      expect(mainDestinationLabelsFor(locale.code), hasLength(13));
      expect(mainDestinationLabelsFor(locale.code),
          isNot(contains('QuickTagCloud')));
    }
    expect(quickTagUi('zh-CN')['catalog'], '全部资料库');
  });
}

File: src/quicktag-source-names.test.ts
import { describe, expect, it } from "vitest";
import fs from "node:fs";
import { ONLINE_GALLERY_SOURCES, onlineGallerySourceInfo } from "./online-gallery";
import { PROMPT_CODEX_BOOKS } from "./prompt-codex";
import { ACTIVE_TABS } from "./app/navigation";
import { desktopUiText, SUPPORTED_APP_LANGUAGES } from "./i18n";
import labels from "../shared/quicktag-ui.json";

describe("QuickTagCloud and nai4.top are identified separately", () => {
  it("keeps QuickTagCloud in the existing gallery, not a new main tab", () => {
    expect(ONLINE_GALLERY_SOURCES.filter(source => source.id === "quicktag")).toHaveLength(1);
    expect(onlineGallerySourceInfo("quicktag").label).toBe("QuickTagCloud");
    expect(onlineGallerySourceInfo("quicktag").siteUrl).toBe("https://novelai.quicktagcloud.com");
    expect([...ACTIVE_TABS]).not.toContain("quicktag");
  });
  it("retains all personal Codex books at nai4.top", () => {
    expect(PROMPT_CODEX_BOOKS).toHaveLength(3);
    expect(PROMPT_CODEX_BOOKS.every(book => new URL(book.sourceUrl).hostname === "nai4.top")).toBe(true);
  });
  it("identifies the personal Codex source and QuickTag collections in every language", () => {
    for (const { code } of SUPPORTED_APP_LANGUAGES) {
      const title = desktopUiText(code, "promptCodex.enabled");
      expect(title).toContain("nai4.top");
      expect(title).not.toContain("QuickTagCloud");
      expect(labels[code].hint).toContain("QuickTagCloud");
    }
    expect(labels["zh-CN"].catalog).toBe("全部资料库");
    expect(labels["en-US"].catalog).toBe("All collections");
  });
  it("uses the adapter's name without the misleading Codex suffix", () => {
    const source = fs.readFileSync(new URL("./AitagGallery.tsx", import.meta.url), "utf8");
    expect(source).not.toContain('QuickTagCloud ·');
    expect(source).toContain('return onlineGallerySourceInfo(source).label;');
    expect(source).toContain('<span>{onlineGallerySourceInfo(item.source).label}</span>');
  });
});
